test_port_allocation.sh

```
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Test script to verify port allocation for multiple bot instances
+# This script simulates multiple concurrent bot launches
+
+echo "Testing port allocation for multiple bot instances..."
+echo "======================================================="
+
+# Source the run_bot.sh to get access to the port functions
+source ./run_bot.sh > /dev/null 2>&1
```

Shellcheck (0.11.0)

[info] Not following: ./run_bot.sh was not specified as input (see shellcheck -x).

(SC1091)